diff --git a/node_modules/tmx-parser/index.js b/node_modules/tmx-parser/index.js
index a51b544..96c5846 100644
--- a/node_modules/tmx-parser/index.js
+++ b/node_modules/tmx-parser/index.js
@@ -686,7 +686,17 @@ function parse(content, pathToFile, cb) {
 }
 
 function defaultReadFile(name, cb) {
-  fs.readFile(name, { encoding: 'utf8' }, cb);
+  var isBrowser = typeof window !== 'undefined';
+  if (isBrowser) {
+    fetch(name)
+      .then(response => response.text())
+      .then(data => {
+        cb(undefined, data);
+      })
+      .catch(e => cb(e));
+  } else {
+    fs.readFile(name, { encoding: 'utf8' }, cb);
+  }
 }
 
 function parseFile(name, cb) {
